test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -449,15 +500,536 @@ def test_last_used_time(self):
+        exec_vnode = status[0]['exec_vnode']
+        node1 = exec_vnode.split(':')[0][1:]
+        jid = self.submit_job(
+            1, {'Resource_List.select': '1:ncpus=1',
+                'Resource_List.place': 'excl'})
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid, extend='x')
+        jid = self.submit_job(
+            1, {'Resource_List.select': '1:ncpus=1',
+                'Resource_List.place': 'excl'})
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid, extend='x')
+        status = self.server.status(JOB, id=jid, extend='x')
+        exec_vnode = status[0]['exec_vnode']
+        node2 = exec_vnode.split(':')[0][1:]
+        self.assertEqual(node1, node2)
+
+    @timeout(1000)
+    def test_power_ramp_down_nodes(self):
+        """
+        Test power hook will ramp down the nodes if power_ramp_rate_enable
+        is enabled and node_idle_limit is reached.
+        """
+        self.setup_power_ramp_rate()
+        a = {"power_ramp_rate_enable": True, 'node_idle_limit': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 60}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        t = int(time.time())
+        self.logger.info("Waiting for 12 mins to ramp down all the nodes")
+        time.sleep(720)
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.expect(NODE, {'state': 'sleep'}, id=n['id'])
+        self.cleanup_power_ramp_rate()
+
+    @timeout(1000)
+    def test_power_ramp_down_max_concurennt_nodes(self):
+        """
+        Test power hook will ramp down only the number of
+        max_concurrent nodes specified in conf file per hook run.
+        """
+        self.setup_power_ramp_rate()
+        a = {"power_ramp_rate_enable": True, 'node_idle_limit': '10',
+             'max_concurrent_nodes': '2'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 60}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        t = int(time.time())
+        self.logger.info("Waiting for 90 secs to ramp down 2 nodes")
+        time.sleep(90)
+        a = {'enabled': 'False'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        i = 0
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                if n['state'] == 'sleep':
+                    i = i + 1
+        print i
+        if i != 2:
+            self.assertFalse("2 nodes are not in sleep state")
```

Comment:
As above, use a single expect() or list comprehension with a len().  Also, remove the print i